File: go/difference-of-squares/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
    // The sum of the first n positive integers is given by n(n+1)/2
	return n*n*(n+1)*(n+1)/4
}

func SumOfSquares(n int) int {
	// The sum of the squares of the first n positive integers is given by n(n+1)(2n+1)/6
    return n*(n+1)*(2*n+1)/6
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: javascript/difference-of-squares/difference-of-squares.js
export class Squares {
  constructor(value) {
    this._value = value;
  }

  get sumOfSquares() {
    return this._value * (this._value + 1) * (2 * this._value + 1) / 6;
  }

  get squareOfSum() {
    return ((this._value * (this._value + 1)) / 2) ** 2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}
